packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -311,7 +311,13 @@ Plugin _pluginFromPackage(String name, Uri packageRoot) {
   if (!globals.fs.isFileSync(pubspecPath)) {
     return null;
   }
-  final dynamic pubspec = loadYaml(globals.fs.file(pubspecPath).readAsStringSync());
+  dynamic pubspec;
+
+  try {
+    pubspec = loadYaml(globals.fs.file(pubspecPath).readAsStringSync());
+  } on YamlException {
+    // Do nothing, potentially not a plugin.
```

Comment:
I'm not familiar with this code, should we printError here, or is this expected to run into `YamlExceptions`s?

Reply:
At this point we're still discovering plugins - I think it should definitely be non-fatal, but I'm not really sure how actionable an error parsing the yaml would be. I will add it to the trace so it gets surfaced somehow

Reply:
Yeah, trace makes more sense actually